JASP-Desktop/widgets/filemenu/filemenu.cpp

```diff
@@ -408,12 +408,22 @@ void FileMenu::setSyncRequest(const QString& path)
 	}
 }
 
-bool FileMenu::checkSyncFileExists(const QString &path)
+bool FileMenu::checkSyncFileExists(const QString &path, bool waitForExistence)
 {
 	if (path.startsWith("http"))
 		return true;
 
 	bool exist = false;
+	int maxExistTrials = waitForExistence ? 5 : 1;
+	int existTrials = 1;
+
+	while (!QFileInfo::exists(path) && existTrials < maxExistTrials)
```

Comment:
You could put the `existTrials++` in the `while` check directly right?

Reply:
That's right.
But I had to think really carefully about the condition 
> while (!QFileInfo::exists(path) && existTrials < maxExistTrials)
that I was already happy to understand that it was the right one...

Reply:
I can imagine :p 
while loops are scary!